docs/api.md

```diff
@@ -30,6 +30,21 @@ var html = katex.renderToString("c = \\pm\\sqrt{a^2 + b^2}", {
 // '<span class="katex">...</span>'
 ```
 
+## Persistent Macros
+
+KaTeX documentation tells the author that `\gdef` will create a macro that persists between KaTeX fields. In order to fulfill that promise, you must make certain provisions. Say that a parser has dropped from the sky with an array of DOM elements that contain math. You could then write this code:
```

Comment:
Here's a proposed second draft:

```suggestion
If you want to define macros within one math element that can be used in other math elements, you need to do two things:
1. Create one shared `macros` object that you pass into every call to `katex.render` or `katex.renderToString`. (Do not create a fresh `macros` object for each call.)
2. In your math content, use `\gdef` to define your macros (so that they are defined in the global group, not just the local group of that particular math mode).

For example, suppose that you have an array `mathElements` of DOM elements that contain math. Then you could then write this code:
```

Reply:
> **If** you want to define macros within one math element that can be used in other math elements

I'd like to keep consistency between what we say here and what we say in  `supported.md`. `supported.md` does not have a qualifying "if". It says that `\gdef` will persist, full stop.  Since security makes that uncertain, I want to revise `supported.md` before this PR goes anywhere.

 I'll use much of your draft here, but I may revise parts of it.  Revision coming later today.

Reply:
Sounds good.  I just found it weird to be referring to another part of the documentation, without even a link, as it's assuming a lot of the reader (what if they read this part before the other part?).  Perhaps the documentation that talks about `\gdef` should also refer to this section, so the proviso is linked directly from the potential offender.

Reply:
> without even a link

Excellent point. I will definitely write a link into this PR.

Not a reverse link, though. An author in, say [Observable](https://observablehq.com/), does not need to be bothered with the machinery behind the scenes. That's the sort of person I'm targeting in `supported.md`